modules/core/src/main/java/org/apache/ignite/IgniteSystemProperties.java

```diff
@@ -1177,61 +1678,86 @@
      * which leads to extra memory consumption, otherwise we use view on the
      * list of cluster nodes to reduce memory consumption on redundant data structures.
      */
+    @SystemProperty("Disables memory optimization: BitSets instead of HashSets to store " +
+        "partitions. When number of backups per partion is > IGNITE_AFFINITY_BACKUPS_THRESHOLD we use HashMap to " +
+        "improve contains() which leads to extra memory consumption, otherwise we use view on the list of cluster " +
+        "nodes to reduce memory consumption on redundant data structures")
     public static final String IGNITE_DISABLE_AFFINITY_MEMORY_OPTIMIZATION = "IGNITE_DISABLE_AFFINITY_MEMORY_OPTIMIZATION";
 
     /**
      * Limit the maximum number of objects in memory during the recovery procedure.
      */
+    @SystemProperty(value = "Limit the maximum number of objects in memory during the recovery procedure",
+        type = Integer.class)
     public static final String IGNITE_RECOVERY_SEMAPHORE_PERMITS = "IGNITE_RECOVERY_SEMAPHORE_PERMITS";
 
     /**
      * Maximum size of history of server nodes (server node IDs) that ever joined to current topology.
      */
+    @SystemProperty(value = "Maximum size of history of server nodes (server node IDs) that ever joined " +
+        "to current topology", type = Integer.class, defaults = "" + DFLT_NODE_IDS_HISTORY_SIZE)
     public static final String IGNITE_NODE_IDS_HISTORY_SIZE = "IGNITE_NODE_IDS_HISTORY_SIZE";
 
     /**
      * Maximum number of diagnostic warning messages per category, when waiting for PME.
      */
+    @SystemProperty(value = "Maximum number of diagnostic warning messages per category, when waiting for PME",
+        type = Integer.class, defaults = "" + DFLT_DIAGNOSTIC_WARN_LIMIT)
     public static final String IGNITE_DIAGNOSTIC_WARN_LIMIT = "IGNITE_DIAGNOSTIC_WARN_LIMIT";
 
     /**
      * Flag to enable triggering failure handler for node if unrecoverable partition inconsistency is
      * discovered during partition update counters exchange.
      */
+    @SystemProperty("Enables triggering failure handler for node if unrecoverable " +
+        "partition inconsistency is discovered during partition update counters exchange")
     public static final String IGNITE_FAIL_NODE_ON_UNRECOVERABLE_PARTITION_INCONSISTENCY =
         "IGNITE_FAIL_NODE_ON_UNRECOVERABLE_PARTITION_INCONSISTENCY";
 
     /**
      * Allow use composite _key, _val columns at the INSERT/UPDATE/MERGE statements.
      */
+    @SystemProperty("Allow use composite _key, _val columns at the INSERT/UPDATE/MERGE statements")
     public static final String IGNITE_SQL_ALLOW_KEY_VAL_UPDATES = "IGNITE_SQL_ALLOW_KEY_VAL_UPDATES";
```

Comment:
The default value is `false`. Should we mentioned it?

Reply:
Yes. It will be mentioned automatically by the help printer.
Please, take a look into CommandLineStartup.java#452